### ISSUE TITLE
BP question

### ISSUE THREAD

**Issue by GuusH** [sni/Thruk#395]
Hi,

It looks like the state of a BP behaves differently from what I was expecting. That doesn't necessarily mean it's wrong, could also just be me not getting it :)
I created a really simple BP with 3 nodes and the "at_least(2,1)" function on it. What I was expecting was that when 1 node goes CRIT, the BP should go WARN. Similarly when 2 nodes go CRIT, the BP should go CRIT. It does this, but it does it also when nodes are in WARN state. So:
1 node WARN -> BP goes WARN
2 nodes WARN -> BP goes CRIT
and I was expecting
2 nodes WARN -> BP goes WARN

By removing the difference between the various non-OK states of the nodes, a lot of the logic that otherwise would be needed can be done without (how do you otherwise distinguish between 1 node WARN or 1 node CRIT? according to my logic both would set the BP to WARN). But then again, this is a business process, so maybe it's not necessary to make this distinction.

Bottom line: how do I create a BP that does not go CRIT but stays at WARN when 2 or more of the nodes are WARN (and there are no CRITs)? Thanks!

Guus
